WordPress/WordPressTest/WPCrashLoggingDataProviderTests.swift

```diff
@@ -0,0 +1,62 @@
+import XCTest
+import AutomatticTracks
+
+@testable import WordPress
+
+final class WPCrashLoggingDataProviderTests: XCTestCase {
+
+    // MARK: - Testing Log Error
+
+    func testReadingTrackUserInMainThread() throws {
+        // Given
+        let dataProvider = self.makeCrashLoggingDataProvider()
+
+        // When
+        let user = try XCTUnwrap(dataProvider.currentUser)
+
+        // Then
+        XCTAssertEqual(user.userID, "\(Constants.defaultAccountID)")
+        XCTAssertEqual(user.username, Constants.defaultAccountUsername)
+        XCTAssertEqual(user.email, Constants.defaultAccountEmail)
+    }
+
+    func testReadingTrackUserInBackgroundThread() async {
+        let dataProvider = self.makeCrashLoggingDataProvider()
+        await withThrowingTaskGroup(of: Void.self) { group in
+            for _ in 1...1000 {
```

Comment:
What is the rationale for adding 1,000 tasks? Is there something about the amount of tasks that affects the test behavior?

Reply:
@mokagio Running 1000 tasks concurrently in the test is an attempt to reveal potential thread safety issues. The choice of 1000 is arbitrary; it's large enough to stress the API but allows the test to be completed in a reasonable time.